meta-refkit-extra/doc/robotics-uarm-metal.rst

```diff
@@ -0,0 +1,293 @@
+           print("SEQ_NUM: %s %s" % (seq_num, unpacked))
+           if seq_num > 0:
+               self.loop.create_task(self.robot.move_servos(unpacked[0],
+                                                            unpacked[1],
+                                                            unpacked[3],
+                                                            unpacked[4],
+                                                            unpacked[11]))
+
+   class JointStateStreamProtocol(asyncio.Protocol):
+       """Implements joint states streaming protocol."""
+
+       def __init__(self, loop, robot):
+           self.loop = loop
+           self.robot = robot
+
+       def connection_made(self, transport):
+           print("A client connected to Joint State Stream")
+           self.robot.stream_joint_state(transport)
+
+   def main():
+       """Entry point."""
+
+       loop = asyncio.get_event_loop()
+       robotarm = RobotArm(loop)
+
+       proto_factory = lambda: JointPositionStreamProtocol(loop, robotarm)
+       joint_pos_coro = loop.create_server(proto_factory, '0.0.0.0', 11000)
+
+       proto_factory = lambda: JointStateStreamProtocol(loop, robotarm)
+       joint_state_coro = loop.create_server(proto_factory, '0.0.0.0', 11002)
+
+       joint_pos_server = loop.run_until_complete(joint_pos_coro)
+       joint_state_server = loop.run_until_complete(joint_state_coro)
+
+       # Serve requests until Ctrl+C is pressed
+       print('Serving on {}'.format(joint_pos_server.sockets[0].getsockname()))
+       print('Serving on {}'.format(joint_state_server.sockets[0].getsockname()))
+       try:
+           loop.run_forever()
+       except KeyboardInterrupt:
+           pass
+
+       # Close the server
+       joint_pos_server.close()
+       joint_state_server.close()
+       loop.run_until_complete(asyncio.gather(joint_pos_server.wait_closed(),
+                                              joint_state_server.wait_closed()))
+       robotarm.deinit()
+       loop.close()
+
+   if __name__ == "__main__":
+       main()
+
+3. Set up ROS environment on the device::
+
+     export ROS_ROOT=/opt/ros
+     export ROS_DISTRO=indigo
+     export ROS_PACKAGE_PATH=/opt/ros/indigo/share
+     export PATH=$PATH:/opt/ros/indigo/bin
+     export LD_LIBRARY_PATH=/opt/ros/indigo/lib
+     export PYTHONPATH=/opt/ros/indigo/lib/python3.5/site-packages
+     export ROS_MASTER_URI=http://localhost:11311
+     export CMAKE_PREFIX_PATH=/opt/ros/indigo
+     touch /opt/ros/indigo/.catkin
+
+4. Launch all the needed ROS nodes with the command::
+
+     roslaunch uarmmetal_support refkit-uarm.launch robot_ip:=127.0.0.1 sim:=false
+
+Now your mechanical arm can be controlled through MoveIt's Move Group Interface.
+It's possible to control the arm interactively  with a GUI ROS node installed
+on your desktop. For that copy the source code of the packages `uarmmetal-support`
+and `uarmmetal-moveit-configs` to your catkin workspace on the desktop as
```

Comment:
I know I may sound ignorant but I had no idea what a catkin workspace was... How about adding a URL (e.g http://wiki.ros.org/catkin) in the text that would provide a pointer to the other less-informed out there ;)